### ISSUE TITLE
ExplanationDashboard code does not generate views

### ISSUE THREAD

**Issue by ABMANI** [Azure/MachineLearningNotebooks#506]
Hi Team,

When I run the ExplanationDashboard code in the jupyter notebook as shown below - 
ExplanationDashboard(global_explanation, model, x_test).

The code runs but gives the following output without generating any visuals as specified - 
<azureml.contrib.explain.model.visualize.ExplanationDashboard.ExplanationDashboard at 0x173b25120b8>

I tried the code on the multiclass example shared using iris data after having followed all of the steps mentioned. 

TIA​
![ExplanationDashboard](https://user-images.githubusercontent.com/42696249/62106072-db593680-b2c1-11e9-813e-5d7b8807ef5b.PNG)

**Comment by rihorn2** [comment]
Hi Abhinav,
To get some context on your environment, could you post the versions of the python library being used, as well as the jupyter widget. Please run ` pip show azureml-explain-model` ,  `pip show azureml-contrib-explain-model`  and `jupyter nbextension list`. Also, after running the notebook the first time, please be sure to restart the kernel and refresh the browser page.
Thank you,
Brandon

**Comment by andreas-wilm** [comment]
Hi there,

I have the exact same problem. I ran 

    ! sudo /anaconda/envs/azureml_py36/bin/jupyter nbextension install --py --sys-prefix azureml.contrib.explain.model.visualize
    ! sudo /anaconda/envs/azureml_py36/bin/jupyter nbextension enable --py --sys-prefix azureml.contrib.explain.model.visualize

from within the ML Notebook first and restarted the kernel. Only the following extensions show up:

```
! jupyter nbextension list

Known nbextensions:
  config dir: /anaconda/envs/azureml_py36/etc/jupyter/nbconfig
    notebook section
      azureml/kernel  enabled 
      - Validating: OK
      azureml_widgets/extension  enabled 
      - Validating: OK
      microsoft-mli-widget/extension  enabled 
      - Validating: OK
      microsoft-mli-widget/index  enabled 
      - Validating: OK
      nbdime/index  enabled 
      - Validating: OK
      jupyter-js-widgets/extension  enabled 
      - Validating: OK

```

**Comment by rihorn2** [comment]
@andreas-wilm This is the correct list output, microsoft-mli-widget is the name of the dashboard widget. The azureml_py36 conda environment should then be able to show the widget. You need the jupyter kernel to be running in this environment to have the widget appear. Use conda install nb_conda to enable setting the conda environment in jupyter notebooks (needs python 3.6 or later). Refresh the browser page after setting the environment. Please let me know if the issue persists.

**Comment by andreas-wilm** [comment]
Not sure if it was kernel restarting or the added `conda install nb_conda`, but the widget displays okay now. Thanks!

**Comment by jimwill3** [comment]
my environment (local laptop) shows these installs/versions (widget is not displaying), restarting kernel and/or browser hasn't resolved it so far. python version is 3.6.8
Known nbextensions:
  config dir: C:\anaconda3\envs\myml\etc\jupyter\nbconfig
    notebook section
      azureml/kernel enabled 
      azureml_widgets/extension enabled 
      microsoft-mli-widget/extension enabled 
      microsoft-mli-widget/index enabled 
      plotlywidget/extension enabled 
      jupyter-js-widgets/extension enabled 
      nb_conda/main enabled 
    tree section
      nb_conda/tree enabled 


      - Validating: ok
      - Validating: ok
      - Validating: ok
      - Validating: ok
      - Validating: ok
      - Validating: ok
      - Validating: ok
      - Validating: ok
and for the azureml libraries:
  Name: azureml-explain-model
Version: 1.0.60
Summary: UNKNOWN
Home-page: https://docs.microsoft.com/en-us/azure/machine-learning/service/
Author: Microsoft Corp
Author-email: None
License: https://aka.ms/azureml-sdk-license
Location: c:\anaconda3\envs\myml\lib\site-packages
Requires: scikit-learn, numpy, shap, scipy, pandas
Required-by: azureml-contrib-explain-model
Name: azureml-contrib-explain-model
Version: 1.0.60
Summary: UNKNOWN
Home-page: https://docs.microsoft.com/en-us/azure/machine-learning/service/
Author: Microsoft Corp
Author-email: None
License: Proprietary https://aka.ms/azureml-preview-sdk-license 
Location: c:\anaconda3\envs\myml\lib\site-packages
Requires: azureml-explain-model
Required-by:

**Comment by imatiach-msft** [comment]
Hi @jimwill3 ,
Sorry about the trouble you are having.  Based on your output you are hitting a known issue that was recently fixed and should be in the next update – the jupyter-js-widgets is showing up after the microsoft-mli-widget:

microsoft-mli-widget/extension enabled
microsoft-mli-widget/index enabled
plotlywidget/extension enabled
jupyter-js-widgets/extension enabled

Running this command and restarting your kernel should fix the issue for you:
jupyter nbextension enable --py widgetsnbextension

This should be fixed in the next update so you won’t need to run the additional command.

**Comment by jimwill3** [comment]
thanks. I tried that but have very inconsistent results. the sample explain-simple-regression-local fails to visualize, however the sample explain-multiclass-classification-local does visualize (that's the only sample to visualize).

the output from !jupyter nbextensions list in the notebooks is:

Known nbextensions:
  config dir: C:\Users\jiwillia\.jupyter\nbconfig
    notebook section
      jupyter-js-widgets/extension enabled 
  config dir: C:\anaconda3\envs\myml\etc\jupyter\nbconfig
    notebook section
      azureml/kernel enabled 
      azureml_widgets/extension enabled 
      microsoft-mli-widget/extension enabled 
      microsoft-mli-widget/index enabled 
      plotlywidget/extension enabled 
      jupyter-js-widgets/extension enabled 
      nb_conda/main enabled 
    tree section
      nb_conda/tree enabled

**Comment by imatiach-msft** [comment]
@jimwill3 you might need to refresh the browser after the changes, can you try the notebook that failed again?  Otherwise, there might be issues with how your python environment is setup.

**Comment by jimwill3** [comment]
refreshing the browser took care of it. thanks - looks fine now. 
Jim

**Comment by rihorn2** [comment]
If refreshing the browser did not work, you can replace the contents of c:\anaconda3\envs\myml\etc\jupyter\nbconfig\notebook.d\microsoft-mli-widget.json with 

```
{
  "load_extensions": {
    "jupyter-js-widgets/extension": true,
    "microsoft-mli-widget/extension": true,
    "microsoft-mli-widget/index": true
  }
}
```
This is the fix rolling out in the next deployment, making that modification and restarting kernel and refreshing the browser page should address the issue.

**Comment by jimwilliam** [comment]
what was the next deployment version number? I just upgraded to 1.0.69 and the issue has resurfaced.
Jim

**Comment by jimwill3** [comment]
Please ignore above. somehow my explain packages updated to 1.0.69 but azureml.core remained at 1.0.60. Once i took care of that things are working fine.
Jim

**Comment by SameerKSharma** [comment]
HI
I am running the Azure example explainer notebooks and i cannot see the visuals (refreshed, followed above steps etc)
Notebook: advanced-feature-transformations-explain-local
Note all example notebooks with explainer visual have same issue

<azureml.contrib.interpret.visualize.ExplanationDashboard.ExplanationDashboard at 0x2be632c9320>



Name: azureml-explain-model
Version: 1.0.69
Summary: UNKNOWN
Home-page: https://docs.microsoft.com/en-us/azure/machine-learning/service/
Author: Microsoft Corp
Author-email: None
License: https://aka.ms/azureml-sdk-license
Location: c:\users\user\anaconda3\envs\myenv\lib\site-packages
Requires: shap, azureml-interpret
Required-by:

Name: azureml-contrib-explain-model
Version: 1.0.65
Summary: UNKNOWN
Home-page: https://docs.microsoft.com/en-us/azure/machine-learning/service/
Author: Microsoft Corp
Author-email: None
License: Proprietary https://aka.ms/azureml-preview-sdk-license
Location: c:\users\user\anaconda3\envs\myenv\lib\site-packages
Requires: azureml-explain-model
Required-by:

Seems weird here

Traceback (most recent call last):
  File "C:\Users\User\Anaconda3\envs\myenv\Scripts\jupyter-nbextension-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\User\Anaconda3\envs\myenv\lib\site-packages\jupyter_core\application.py", line 267, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "C:\Users\User\Anaconda3\envs\myenv\lib\site-packages\traitlets\config\application.py", line 664, in launch_instance
    app.start()
  File "C:\Users\User\Anaconda3\envs\myenv\lib\site-packages\notebook\nbextensions.py", line 988, in start
    super(NBExtensionApp, self).start()
  File "C:\Users\User\Anaconda3\envs\myenv\lib\site-packages\jupyter_core\application.py", line 256, in start
    self.subapp.start()
  File "C:\Users\User\Anaconda3\envs\myenv\lib\site-packages\notebook\nbextensions.py", line 960, in start
    self.list_nbextensions()
  File "C:\Users\User\Anaconda3\envs\myenv\lib\site-packages\notebook\nbextensions.py", line 943, in list_nbextensions
    data = cm.get(section)
  File "C:\Users\User\Anaconda3\envs\myenv\lib\site-packages\notebook\config_manager.py", line 103, in get
    recursive_update(data, json.load(f))
  File "C:\Users\User\Anaconda3\envs\myenv\lib\json\__init__.py", line 299, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "C:\Users\User\Anaconda3\envs\myenv\lib\json\__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "C:\Users\User\Anaconda3\envs\myenv\lib\json\decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Users\User\Anaconda3\envs\myenv\lib\json\decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

Cheers
Sam

**Comment by v-strudm-msft** [comment]
Thank for the latest post on this issue.  We're glad the earlier reported issues were resolved. The latest issue was posted some time ago and was not addressed. Our apologies. We will close it. Should you still need assistance with this or another issue, please re-post and we will respond promptly.
